apps/dashboard/lib/trpc/routers/authorization/roles/llm-search/index.ts

```diff
@@ -0,0 +1,20 @@
+import { env } from "@/lib/env";
+import { ratelimit, requireUser, requireWorkspace, t, withRatelimit } from "@/lib/trpc/trpc";
+import OpenAI from "openai";
+import { z } from "zod";
+import { getStructuredSearchFromLLM } from "./utils";
+
+const openai = env().OPENAI_API_KEY
+  ? new OpenAI({
+      apiKey: env().OPENAI_API_KEY,
+    })
+  : null;
+
+export const rolesLlmSearch = t.procedure
+  .use(requireUser)
+  .use(requireWorkspace)
+  .use(withRatelimit(ratelimit.read))
+  .input(z.object({ query: z.string() }))
```

Comment:
you'll want stricter limits on this rather than just a type check. a length check for sure so we don't blast through tokens on long strings

Reply:
Sounds good. ENG-1805